packages/vscode-extension/src/webview/views/ActivateLicenseView.tsx

```diff
@@ -0,0 +1,108 @@
+import { useRef, useState } from "react";
+import { FieldValues, SubmitHandler, useForm } from "react-hook-form";
+import "./ActivateLicenseView.css";
+import { VSCodeProgressRing } from "@vscode/webview-ui-toolkit/react";
+import Button from "../components/shared/Button";
+import { useProject } from "../providers/ProjectProvider";
+import { useModal } from "../providers/ModalProvider";
+import { ActivateDeviceResult } from "../../common/Project";
+
+export function ActivateLicenseView() {
+  const { project } = useProject();
+  const { closeModal } = useModal();
+  const { register, handleSubmit } = useForm();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [disableSubmit, setDisableSubmit] = useState(true);
+  const [wasRejected, setWasRejected] = useState(false);
+  const [wasEnoughSeats, setWasEnoughSeats] = useState(true);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const onSubmit: SubmitHandler<FieldValues> = (e, data) => {
+    setIsLoading(true);
+
+    const activationPromise = project.activateLicense(data?.target[0].value);
+
+    activationPromise.then((activationResult) => {
+      if (activationResult === ActivateDeviceResult.succeeded) {
+        closeModal();
+      } else {
+        if (activationResult === ActivateDeviceResult.notEnoughSeats) {
+          setWasEnoughSeats(false);
+        }
+        setWasRejected(true);
+      }
+      setIsLoading(false);
+    });
+  };
+
+  const onChange = () => {
+    const newInput = inputRef.current?.value;
+    if (newInput) {
+      const isValidKey = /^[0-9A-F]{4}(?:-[0-9A-F]{4}){7}$/.test(newInput);
+      setDisableSubmit(!isValidKey);
+      return;
+    }
+    setDisableSubmit(true);
+  };
+
+  return (
+    <form className="container" onSubmit={handleSubmit(onSubmit)}>
+      <div className="info-row">
+        {!wasRejected && (
+          <div className="info-text">
+            You can find your license key on the Radon IDE customer portal (
+            <a href="https://portal.ide.swmansion.com/" target="_blank" rel="noopener noreferrer">
+              link
+            </a>
+            ) If you don't have a license, you can purchase it{" "}
+            <a href="https://ide.swmansion.com/pricing" target="_blank" rel="noopener noreferrer">
+              here
+            </a>
+            .
+          </div>
+        )}
+        {wasRejected && wasEnoughSeats && (
+          <div className="error-text">
+            Unable to verify the key. Please ensure your license key is correct. Check this{" "}
```

Comment:
We should either distinguish the case where license servers cannot be reached (e.g. no connection) vs the case when license key is invald or make this message inform the user that network connection is needed for activation.

Reply:
done

Reply:
As pointed out earlier, we need a way to tell whether someone has put in a correct key vs when there was some possibly random issue validating it (i.e. servers are down)

In the first scenario we can write:
> Provided key is not a valid license key. You can find your key on the [Radon IDE customer portal]. Please follow the [license activation instructions] if you need further assistance.

For the case of some generic error in which we know that this isn't key validation response from the server we could write:
> We were unable to verify the license key at this point. Please consult the [license activation instructions] and try again once you check the provided key is valid.